avm/res/compute/virtual-machine/tests/e2e/linux.defaults/main.test.bicep

```diff
@@ -37,6 +37,7 @@ module nestedDependencies 'dependencies.bicep' = {
   params: {
     location: resourceLocation
     virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    maintenanceConfigurationName: 'dep-${namePrefix}-mc-${serviceShort}'
```

Comment:
Should only be set in the WAF tests as this is an optional value. For the default tests, if a PSRule test fails for a parameter that requires a user input (as opposed to e.g., a boolean) then we should exclude it via the PS Rule min configuration file (with a comment to explain why)

Reply:
The same is true for all other default parameters

Reply:
but the name is not optional, but required

Reply:
Hey @rahalan,
I think we might be talking past each other. The `maintenanceConfigurationName` here is mandatory yes - but only because you deploy a maintenanceConfiguration in the dependencies. However, I question if that is necessary as you only deploy it becausse you set 3 new parameters in the defaults test
```
      patchMode: 'AutomaticByPlatform'
      bypassPlatformSafetyChecksOnUserSchedule: true
      maintenanceConfigurationResourceId: nestedDependencies.outputs.maintenanceConfigurationResourceId
```
which I'd argue should probably not be set in a test that wants to test if the module works if you only set required parameters 😉

Reply:
@AlexanderSehr without those parameters you cannot set a maintenanceConfiguration. And I deploy the config, because I need to set in according to WAF.

Reply:
done